pkg/data/domain/appcatalog/spec.go

```diff
@@ -4,16 +4,70 @@ import (
 	"context"
 
 	applicationv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// AppCatalog abstracts away the custom resource so it can be returned as a runtime
+// object or a typed custom resource.
+type AppCatalog struct {
+	CR      *applicationv1alpha1.AppCatalog
+	Entries *applicationv1alpha1.AppCatalogEntryList
+}
+
+// Collection wraps a list of appcatalogs.
+type Collection struct {
+	Items []AppCatalog
+}
+
 // GetOptions are the parameters that the Get method takes.
 type GetOptions struct {
-	Name string
+	LabelSelector string
```

Comment:
What do you think about using the `labels.Selector` type? This way we require the command (or the caller) to parse and validate the input, in the same way we do with command line flags.

Reply:
Thanks @axbarsan I like that so the input can be validated. Done.